src/anyio/_backends/_asyncio.py

```diff
@@ -2046,6 +2048,7 @@ def run_sync_from_thread(
     ) -> T_Retval:
         @wraps(func)
         def wrapper() -> None:
+            sniffio.current_async_library_cvar.set("asyncio")
```

Comment:
I would feel safer if this was inside the `try` block, just in case this call ever fails for whatever reason. At least that way the waiter of the future won't be left waiting forever.